packages/ui/primitives/signature-pad/signature-pad.tsx

```diff
@@ -91,31 +75,33 @@ export const SignaturePad = ({
 
     const point = Point.fromEvent(event, DPI, $el.current);
 
-    if (point.distanceTo(points[points.length - 1]) > 5) {
-      const newPoints = [...points, point];
-
-      setPoints(newPoints);
+    if (point.distanceTo(currentLine[currentLine.length - 1]) > 5) {
+      setCurrentLine([...currentLine, point]);
 
+      // Update the canvas here to draw the lines
       if ($el.current) {
         const ctx = $el.current.getContext('2d');
 
         if (ctx) {
-          ctx.restore();
+          ctx.clearRect(0, 0, $el.current.width, $el.current.height);
 
-          ctx.imageSmoothingEnabled = true;
-          ctx.imageSmoothingQuality = 'high';
```

Comment:
Any reason to remove these? They help maintain image quality 🙌

Reply:
I apologize. Done by mistake.

Reply:
OKadding back and commiting changes